xarray/backends/netCDF4_.py

```diff
@@ -210,6 +211,19 @@ def _extract_nc4_variable_encoding(variable, raise_on_invalid=False,
         if chunks_too_big or changed_shape:
             del encoding['chunksizes']
 
+    var_has_unlim_dim = any(dim in unlimited_dims for dim in variable.dims)
+    if var_has_unlim_dim:
```

Comment:
Add a comment saying "unlimited dimensions" cannot be contiguous. 

Can you combine the two if's?

Reply:
yes